src/main/java/co/cask/hydrator/salesforce/authenticator/AuthResponse.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.authenticator;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Objects;
+
+/**
+ * Oauth2 response from salesforce server
+ */
+@VisibleForTesting
+public class AuthResponse {
+  @SerializedName("access_token")
+  private final String accessToken;
+  @SerializedName("instance_url")
+  private final String instanceUrl;
+  private final String id;
+  @SerializedName("token_type")
+  private final String tokenType;
+  @SerializedName("issued_at")
+  private final String issuedAt;
+  private final String signature;
+
+  private AuthResponse(String accessToken, String instanceUrl, String id, String tokenType,
+                       String issuedAt, String signature) {
+    this.accessToken = accessToken;
+    this.instanceUrl = instanceUrl;
+    this.id = id;
+    this.tokenType = tokenType;
+    this.issuedAt = issuedAt;
+    this.signature = signature;
+  }
+
+  public String getAccessToken() {
+    return accessToken;
+  }
+
+  public String getInstanceUrl() {
+    return instanceUrl;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public String getTokenType() {
+    return tokenType;
+  }
+
+  public String getIssuedAt() {
+    return issuedAt;
+  }
+
+  public String getSignature() {
+    return signature;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    AuthResponse that = (AuthResponse) o;
+
+    return (Objects.equals(accessToken, that.getAccessToken()) &&
```

Comment:
Be careful while using Objects.equals because in this method two null objects are equal. Is it expected behavior?

Reply:
I suggest to use fields instead of getters

Reply:
> Be careful while using Objects.equals because in this method two null objects are equal. Is it expected behavior?
yes it is expected.